code/modules/surgery/dental_surgeries.dm

```diff
@@ -77,7 +77,12 @@
 	time = 16
 
 /datum/surgery_step/insert_pill/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-	user.visible_message("[user] begins to wedge \the [tool] in [target]'s [parse_zone(target_zone)].", "<span class='notice'>You begin to wedge [tool] in [target]'s [parse_zone(target_zone)]...</span>")
+	if(istype(tool, /obj/item/weapon/reagent_containers/pill) && tool.parent_type != /obj/item/weapon/reagent_containers/pill)
```

Comment:
Instead of this,just do a strict type check, which means excluding childs. if(type == /obj/item/weapon/reagent_containers/pill)